sdk/core/azure-core/tests/async_tests/test_authentication_async.py

```diff
@@ -7,17 +7,26 @@
 import time
 from unittest.mock import Mock
 
+from requests.models import RequestEncodingMixin
```

Comment:
why are there so many changes to the existing test files?

Reason is:
1. We mock transports a lot, and now that I've added these new properties to the transport, sometimes need to mock the properties
2. I've parameterized p much every test (besides `test_basic_transport.py`). Each test now first runs pipeline transport, then rest